### PR TITLE
feat: add easyssh and go routine.

### LINKED ISSUE #37: Support go routines

### PR DESCRIPTION
**Pull request by appleboy**
fix #37

### CODE REVIEW

File: easyssh/easyssh.go [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+)
+
+// MakeConfig Contains main authority information.
+// User field should be a name of user on remote server (ex. john in ssh john@example.com).
+// Server field should be a remote machine address (ex. example.com in ssh john@example.com)
+// Key is a path to private key on your local machine.
+// Port is SSH server port on remote machine.
+// Note: easyssh looking for private key in user's home directory (ex. /home/john + Key).
+// Then ensure your Key begins from '/' (ex. /.ssh/id_rsa)
+type MakeConfig struct {
```
Comment (tboerger) [inline easyssh/easyssh.go]: Just Config?

File: easyssh/easyssh.go [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+	User     string
+	Server   string
+	Key      string
+	KeyPath  string
+	Port     string
```
Comment (tboerger) [inline easyssh/easyssh.go]: Int?

File: easyssh/easyssh.go [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+		auths = append(auths, ssh.PublicKeys(signer))
+	}
+
+	config := &ssh.ClientConfig{
+		Timeout: ssh_conf.Timeout,
+		User:    ssh_conf.User,
+		Auth:    auths,
+	}
+
+	client, err := ssh.Dial("tcp", ssh_conf.Server+":"+ssh_conf.Port, config)
```
Comment (tboerger) [inline easyssh/easyssh.go]: There is a helper like JoinHostPort

Comment (tboerger) [inline easyssh/easyssh.go]: https://golang.org/pkg/net/#JoinHostPort

File: easyssh/easyssh.go (hunk 2 above)
Comment (appleboy) [inline easyssh/easyssh.go]: https://github.com/appleboy/drone-ssh/blob/master/plugin.go#L64

File: easyssh/easyssh.go (hunk 3 above)
Comment (appleboy) [inline easyssh/easyssh.go]: Done https://github.com/appleboy/drone-ssh/commit/a8cc62009f4c594ffe35b3a9a5f13eb1c0f40fb1

File: easyssh/easyssh.go (hunk 2 above)
Comment (tboerger) [inline easyssh/easyssh.go]: I know, but why transforming to a string?

Comment (appleboy) [inline easyssh/easyssh.go]: @tboerger I think we can change `port` type from `int` to `string` on https://github.com/appleboy/drone-ssh/blob/master/main.go#L54-L58

**Pull request merged**